app/components/submission/ManuscriptMetadata/SubjectAreaDropdown.test.js

```diff
@@ -0,0 +1,120 @@
+import React from 'react'
+import { shallow, mount } from 'enzyme'
+import theme from '@pubsweet/elife-theme'
+
+import SubjectAreaDropdown from './SubjectAreaDropdown'
+
+const onChange = jest.fn()
+
+const MyDropdown = (
+  <SubjectAreaDropdown
+    label="My label"
+    name="My name"
+    onBlur={jest.fn()}
+    onChange={onChange}
+    savedValues={[]}
+    theme={theme}
+  />
+)
+
+describe('SubjectAreaDropdown component', () => {
+  it('renders label & select', () => {
+    const wrapper = shallow(MyDropdown).dive() // dive is necessary because SubjectAreaDropdown is wrapped by withTheme
+    expect(wrapper.contains('My label')).toBe(true)
+    expect(wrapper.find('[inputId="subject-area-select"]').exists()).toEqual(
```

Comment:
What are you testing here? I would see this as a test to see if the SubjectAreaDropdown has an element inside with id `subject-area-select` (implementation specific). I'm not a testing expert, but I usually test behavioral stuff (e.g. given these props, the component displays this text).

Reply:
The whole test is basically 'does this render' correctly? You could use a snapshot test for this. What I've done is say
- is there a label present?
- is there an input field present?

For the second bullet point, just plain ol' `wrapper.find('input')` would be enough. Check if there's an input at all.

But the way `react-select` works under the hood, when you select an option from the dropdown, it creates hidden input fields. So I guess I was just going specific, to see if there was _this specific_ input field present, since I've hard coded `id=subject-area-dropdown` into SubjectAreaSelect

Reply:
...probably unnecessary though :thinking: We'll never have the hidden inputs present, & the main input field _not_ present....

Reply:
Probably a snapshot test would be better? Like I said, I wouldn't test for this. I would just try to test functionality, but this is not a `request changes` comment, just an open discussion, so feel free to keep it like this if you think it makes sense :)

Reply:
Yeah I agree. Seems like a good case for a snapshot because all I really want to say is "is this on the page?" & maybe "does it have this general form?" i.e. is there a label & an input field.

I can understand the resistance to snapshot testing, so I'm happy not to start including snapshots in our codebase. But I reckon having a test which checks that the component renders at all is a good thing & I reckon the first couple of assertions do that job here.

The 3rd assertion is probably misplaced :thinking: 
I guess there should be a test that sets the state (specifically `hasReachedMultiselectLimit=false`) & then checks whether there's a message? What do you think @dryhten?